src/evo/deterministicmns.h

```diff
@@ -40,18 +40,24 @@ class CDeterministicMN
     uint64_t internalId{std::numeric_limits<uint64_t>::max()};
 
 public:
+    static constexpr uint16_t TYPE_REGULAR_MASTERNODE = 0;
+    static constexpr uint16_t TYPE_HIGH_PERFORMANCE_MASTERNODE = 1;
+
+    static constexpr CAmount REGULAR_MASTERNODE_WEIGHT{1};
+    static constexpr CAmount REGULAR_MASTERNODE_COLLATERAL{1000 * COIN};
+    static constexpr CAmount HIGH_PERFORMANCE_MASTERNODE_WEIGHT{4};
+    static constexpr CAmount HIGH_PERFORMANCE_MASTERNODE_COLLATERAL{REGULAR_MASTERNODE_COLLATERAL * CDeterministicMN::HIGH_PERFORMANCE_MASTERNODE_WEIGHT};
+
+    static constexpr uint16_t CURRENT_MN_FORMAT = 0;
+    static constexpr uint16_t MN_TYPE_FORMAT = 1;
+
     CDeterministicMN() = delete; // no default constructor, must specify internalId
-    explicit CDeterministicMN(uint64_t _internalId) : internalId(_internalId)
+    explicit CDeterministicMN(uint64_t _internalId, bool highPerformanceMasternode = false) : internalId(_internalId)
```

Comment:
I'd prefer this param be the above enum I suggested creating

Reply:
@PastaPastaPasta No for this one since it is called here:
`std::make_shared<CDeterministicMN>(newList.GetTotalRegisteredCount(),proTx.nType==CProRegTx::TYPE_HIGH_PERFORMANCE_MASTERNODE);
`

I really don't want to mix CDeterministicMN enum for type and CProRegTx types.